src/InodeCache.hpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2020 Joel Rosdahl and other contributors
+//
+// See doc/AUTHORS.adoc for a complete list of contributors.
+//
+// This program is free software; you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 3 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program; if not, write to the Free Software Foundation, Inc., 51
+// Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+#pragma once
+
+#include "system.hpp"
+
+#include "config.h"
+
+#ifdef INODE_CACHE_SUPPORTED
+
+#  include <stdint.h>
+#  include <string>
+
+class Config;
+class Context;
+struct digest;
+
+class InodeCache
+{
+public:
+  enum class ContentType {
+    binary = 0,
+    code = 1,
```

Comment:
The different types are required to properly handle the pathetic case that the same file is used in more than one role, e.g. given as an afdo file in one run (perhaps by mistake) and then used as a source file in the next run. Reusing the hash result would then be an error since the binary file was never checked for time macros and all of that.

Perhaps precompiled headers should go into their own category as well, by similar reasons? Didn't see that before now.

Reply:
Right. Some comments about this might be in order.